src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -253,7 +257,42 @@ def crud_scenario(self, state: State, id: str, action: str, payload: t.Dict[str,
                 state.assign(_GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR, f"Invalid date ({date_str}).{e}")
                 return
             try:
-                scenario = create_scenario(scenario_config, date, name)
+                do_create = True
+                on_creation = args[3] if len(args) > 3 and isinstance(args[3], str) else None
+                on_creation_function = gui._get_user_function(on_creation) if on_creation else None
+                if callable(on_creation_function):
+                    try:
+                        res = gui._call_function_with_state(
+                            on_creation_function,
+                            [
+                                id,
+                                on_creation,
+                                {
+                                    "config": scenario_config,
+                                    "date": date,
+                                    "name": name,
+                                    "properties": data.get("properties"),
+                                    "before": True,
+                                },
+                            ],
+                        )
+                        if isinstance(res, Scenario):
+                            # everything's fine
+                            state.assign(_GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR, "")
+                            return
+                        if res:
+                            # do not create
+                            on_creation_function = None
+                            do_create = False
+                            state.assign(_GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR, f"{res}")
+                    except Exception as e:  # pragma: no cover
+                        if not gui._call_on_exception(on_creation, e):
+                            _warn(f"on_creation() before: Exception raised in '{on_creation}()':\n{e}")
```

Comment:
"before"?

Reply:
... I thought there would be no second call after all.
If the user wants to tweak the scenario, then he creates it (create_scenario) with the parameters you have provided, then do whatever he wants.